sequencing/ATACSeq/jobSubmission/0_setUp.sh

```diff
@@ -0,0 +1,174 @@
+if [[ ! -d ${METADIR} ]]
+then
+	echo 'METADATA directory does not exist'
+	exit 1
+elif  test -f ${METADIR}/samples.txt; then
+	mapfile -t SAMPLEIDS < ${METADIR}/samples.txt 
+	echo "Sample list found. Number of sample IDs found:" """${#SAMPLEIDS[@]}"""
+else
+	echo 'samples.txt file does not exist in METADIR'
+fi
+
+## Check that the sampleSheet exist
+if [[ -d ${METADIR} ]] && [[ -s "${METADIR}/sampleSheet.csv" ]]
+then
+  SAMPLEIDS2=( $(tail -n +2 ${METADIR}/sampleSheet.csv  | cut -d ',' -f2) )
+  echo "Sample sheet found. Number of sample IDs found:"" ""${#SAMPLEIDS2[@]}"""
+else
+  echo "sampleSheet.csv file does not exist in METADIR."
+fi
+
+## Check if remaining main directories exist and create if not
+dirs=(${RAWDATADIR} ${FASTQCDIR} ${TRIMDIR} ${ALIGNEDDIR} ${PEAKDIR})
+type=("DATA" "FASTQC" "TRIMMED" "ALIGNED" "PEAK CALLED")
+
+for (( x=0; x<${#dirs[@]}; x++ ))
+do
+	if [ -d ${dirs[x]} ] 
+	then 
+		echo ${type[x]} "directory:" ${dirs[x]}
+	elif [ ${dirs[x]} = $RAWDATADIR ]; then 
+		echo ${dirs[x]} "directory does not exist"
+		exit 1
+	else
+		echo "Creating directory: " ${dirs[x]}
+		mkdir -p ${dirs[x]} 
+	fi
+done
+
+## MODULES ##
+
+## Check that the SAMstats environment exists and create it if not
+
+module load Anaconda3
+if [ -d ${ENVDIR} ]
+then 
+	echo "ENCODEQC environment exists"
+  source ${CONDAENV}
+  conda activate ${ENVDIR}
+  echo "List of packages included in conda environment is output in project directory"
+  conda list > "/lustre/projects/Research_Project-MRC190311/ATACSeq/$1/condaEnv.txt"
+  conda deactivate
+else
+	echo "ENCODEQC environment does not exist"
+	echo "Creating environment"
+  source ${CONDAENV}
+	conda create --prefix ${ENVDIR}
+	conda activate ${ENVDIR}
+	echo "Installing SAMstats"
+	conda install -c bioconda samstats
+	conda deactivate
+	echo "ENCODEQC environment has been created and SAMstats installed"
+fi
+
+## Check that pip environment exist and create it if not
+module load Python/3.9.6-GCCcore-11.2.0-bare
+
+if [[ -d ${PIPENV} ]]
+then
+  echo "Pip environment exists."
+  source ${PIPENV}/bin/activate
+  echo "List of packages included in pip environment is output in project directory"
+  pip list > "/lustre/projects/Research_Project-MRC190311/ATACSeq/$1/pipEnv.txt"
+  deactivate
+else
+  echo "Pip environment does not exist."
+  echo "Creating environment."
+  python -m venv ${PIPENV}
+  source ${PIPENV}/bin/activate
+  echo "Installing basic libraries needed in pipeline"
+  pip install Cython
+  pip install cykhash
+  pip install numpy
+  pip install hmmlearn
+  pip install scikit-learn
+  pip install scipy
+  pip install macs3
+  deactivate
+fi
+
+module purge
+
+echo "R v4.2 is required and will be loaded if found"
+## Check that required R version 4.2 exist
+module load R/4.2.1-foss-2022a
+
+echo "R packages will be checked and installed if not found."
+## Check that required R libraries are installed
+Rscript ${SCRIPTDIR}/ATACSeq/preprocessing/setUpRenv.r ${USER}
```

Comment:
We have a similar script in the DNAm part of the pipeline - can we give a consistent filename? array/DNAm/preprocessing/installLibraries.r